extensions/azurecore/src/azureResource/utils.ts

```diff
@@ -347,24 +340,28 @@ export async function makeHttpRequest(account: azdata.Account, subscription: azu
 		validateStatus: () => true // Never throw
 	};
 
+	if (!overrideHost) {
+		serviceUrl = `https://management.azure.com${serviceUrl}`;
+	}
+
 	let response;
 
 	switch (requestType) {
-		case HttpRequestType.GET:
-			response = await axios.get(url, config);
+		case HttpRequestMethod.GET:
+			response = await axios.get(serviceUrl, config);
 			break;
-		case HttpRequestType.POST:
-			response = await axios.post(url, requestBody, config);
+		case HttpRequestMethod.POST:
+			response = await axios.post(serviceUrl, requestBody, config);
 			break;
-		case HttpRequestType.PUT:
-			response = await axios.put(url, requestBody, config);
+		case HttpRequestMethod.PUT:
+			response = await axios.put(serviceUrl, requestBody, config);
 			break;
-		case HttpRequestType.DELETE:
-			response = await axios.delete(url, config);
+		case HttpRequestMethod.DELETE:
+			response = await axios.delete(serviceUrl, config);
 			break;
 	}
 
-	if (response.status !== 200) {
+	if (response.status >= 400) {
```

Comment:
what does this mean?

Reply:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

Based on this doc, I am treating any status with code >= 400 as error codes.

Reply:
Why are you changing this now? This doesn't seem related to your changes. And given that we're not handling any of the other cases (such as the redirects) here I'm not sure we get any benefit from changing this.

Reply:
There is an endpoint that return 201. I am adding this change to handle that edge case

Reply:
Then you should do >= 200 && <= 299 as those are the successful responses.

Reply:
![image](https://user-images.githubusercontent.com/13777222/106668123-bf8dbb00-655e-11eb-8a1d-ea5ad8b46aa8.png)

I think you should only treat 2xx as success, 3xx is for redirection.

Reply:
I was not considering redirections as errors.

Reply:
But unless we handle them then it's not going to be what the user wants - so until we do making them errors makes sense to me.